Sources/Datadog/Core/FeaturesConfiguration.swift

```diff
@@ -173,7 +173,7 @@ extension FeaturesConfiguration {
         if let firstPartyHosts = configuration.firstPartyHosts, !firstPartyHosts.isEmpty {
```

Comment:
it may make sense to not init'ing `URLSessionAutoInstrumentation` in case of empty `userDefinedFirstPartyHosts`, what do you think?

```swift
// pseudo-code
if let firstPartyHosts = sanitized(config.firstPartyHosts) { ... }
```

Reply:
Hmm, I see 👍, but I'm not that sure 💭. 

By continuing on `URLSessionAutoInstrumentation` config we do one more validation (if RUM or Tracing are enabled):
```swift
if let firstPartyHosts = configuration.firstPartyHosts, !firstPartyHosts.isEmpty {
    if configuration.tracingEnabled || configuration.rumEnabled {
        // ... stanitize hosts ...
    } else {
        let error = ProgrammerError(
            description: """
            To use `.track(firstPartyHosts:)` either RUM or Tracing should be enabled.
            """
        )
        consolePrint("\(error)")
    }
}
```
This means that if the user has both things misconfigured, he/she will be warned in one shot, instead of two. At the same time, `firstPartyHosts` misconfiguration is not an expected state, so I wouldn't additionally optimise for it. WDYT?

Reply:
maybe it makes more sense to add this into `URLSessionAutoInstrumentation.init` method, `if userDefinedFirstPartyHosts.isEmpty { return nil }` to avoid unneeded swizzling.

as you said that shouldn't happen often though, you can merge it as it is ✅